File: oop-in-java-vs-python/Car.py
class Vehicle:

 def __init__(self, color, model):
    self.color = color
    self.model = model

class Device:

    def __init__(self):
        self.voltage = 12

class Car(Vehicle, Device):

    wheels = 0

    def __init__(self, color, model, year):
    Vehicle.__init__(self, color, model)
    Device.__init__(self)
    self.year = year

    def add_wheels(self, wheels):
        self.wheels = wheels

    @property
    def voltage(self):
        return self._voltage

    @voltage.setter
    def voltage(self, volts):
        print("Warning: this can cause problems!")
        self._voltage = volts

    @voltage.deleter
    def voltage(self):
        print("Warning: the radio will stop working!")
        del self._voltage

    def __str__(self):
        return f'Car {self.color} : {self.model} : {self.year}'

    def __eq__(self, other):
        return self.year == other.year

    def __lt__(self, other):
        return self.year < other.year

    def __add__(self, other):
        return Car(self.color + other.color,
               self.model + other.model, int(self.year) + int(other.year))

File: oop-in-java-vs-python/Car.java
/**
 * Car class
 * 
 * Used for the Real Python OOP in Java vs Python article
 * 
 * Requires both Vehicle.java and Device.java
 */

public class Car extends Vehicle implements Device {

    private int year; // Year of our car
    private int voltage; // Battery voltage
    private static int wheels; // How many wheels do we have

    /**
     * Car constructor
     */
    public Car(String color, String model, int year) {
        super(color, model);
        this.year = year;
        this.voltage = 12;
    }

    /**
     * Override the interface method.
     */
    @Override
    public int getVoltage() {
        return voltage;
    }

    /**
     * getYear returns the Car's year
     */
    public int getYear() {
        return year;
    }

    /**
     * setYear changes Car's year
     */
    public void setYear(int year) {
        this.year = year;
    }

    /**
     * getWheels returns the number of wheels
     */
    public static int getWheels() {
        return wheels;
    }

    /**
     * setWheels sets the number wheels
     */
    public static void setWheels(int count) {
        wheels = count;
    }

    /**
     * Return a human readable string reoresenting the Car
     */
    public String toString() {
        return "Car: " + getColor() + " : " + getModel() + " : " + getYear();
    }
}